src/ngraph/util.hpp

```diff
@@ -214,12 +217,33 @@ namespace ngraph
     * This utility takes forward-propogation and back-propagation functions
     * and turns them into clone functions where the intermediate values of
     * the forward prop are added to the output of fprop and the input of the bprop
-    * to avoid repeat calcualtions.
+    * to avoid repeat calculations.
     * The last argument is the adjoints coming into the bprop function, the output
     * bprop function will have these nodes as the first N input parameters
     **/
     FpropCache cache_fprop(std::shared_ptr<Function> fprop, std::shared_ptr<Function> bprop);
 
+    // constant folding functions
+    enum class Backend
```

Comment:
If backends have specific constant folding algorithms then they belong in the backend, not in core. This enum should be removed. It is not complete.

Reply:
how about passing a bool variable use_cpu_kernel whose default value is false?